app/gui/view/graph-editor/src/component/edge.rs

```diff
@@ -4,2033 +4,1044 @@ use crate::prelude::*;
+
+        // Calculate closeness to the straight segments.
+        let (linear_x, linear_y) = (dx - radius, dy - radius);
+        let snap_distance = snap_line_width / 2.0;
+        let y_along_vertical = (self.vertical.y() - split_point.y()).abs() <= linear_y;
+        let x_along_horizontal = (self.horizontal.x() - split_point.x()).abs() <= linear_x;
+        let y_near_horizontal = (self.horizontal.y() - split_point.y()).abs() <= snap_distance;
+        let x_near_vertical = (self.vertical.x() - split_point.x()).abs() <= snap_distance;
+
+        // Calculate closeness to the arc.
+        // 1. Find the origin of the circle the arc is part of.
+        // The corner of our bounding box that is immediately outside the arc.
+        let point_outside_arc = Vector2(self.vertical.x(), self.horizontal.y());
+        // The opposite corner of our bounding box, far inside the arc.
+        // Used to find the direction from outside the arc to the origin of the arc's circle.
+        let point_inside_arc = Vector2(self.horizontal.x(), self.vertical.y());
+        let outside_to_inside = point_inside_arc - point_outside_arc;
+        let outside_to_origin =
+            Vector2(radius.copysign(outside_to_inside.x()), radius.copysign(outside_to_inside.y()));
+        let origin = point_outside_arc + outside_to_origin;
+        // 2. Check if the point is on the arc.
+        let input_to_origin = split_point - origin;
+        let distance_squared_from_origin =
+            input_to_origin.x().powi(2) + input_to_origin.y().powi(2);
+        let min_radius = radius - snap_line_width / 2.0;
+        let max_radius = radius + snap_line_width / 2.0;
+        let too_close = distance_squared_from_origin < min_radius.powi(2);
+        let too_far = distance_squared_from_origin > max_radius.powi(2);
+        let on_arc = !(too_close || too_far);
+
+        if y_near_horizontal && x_along_horizontal {
+            // The point is along the horizontal line. Snap its y-value, and draw a corner to it.
+            let snapped = Vector2(split_point.x(), self.horizontal.y());
+            let source_end = self.with_target_end(snapped);
+            let target_end = self.with_source_end(snapped);
+            Some(SplitCorner { source_end, target_end, split_arc: None })
+        } else if x_near_vertical && y_along_vertical {
+            // The point is along the vertical line. Snap its x-value, and draw a corner to it.
+            let snapped = Vector2(self.vertical.x(), split_point.y());
+            let source_end = self.with_target_end(snapped);
+            let target_end = self.with_source_end(snapped);
+            Some(SplitCorner { source_end, target_end, split_arc: None })
+        } else if on_arc {
+            // Find the input point's angle along the arc.
+            let offset_from_origin = split_point - origin;
+            let split_angle = offset_from_origin.y().atan2(offset_from_origin.x());
+            // Split the arc on the angle.
```

Comment:
In the new implementation, a lot of docs from the old implementation are missing, something like this:

```
// node (for example, not to draw the port line above source node).
         //
         // ╭──────────────╮
         // │      ┼ (0,0) │
         // ╰──────────────╯────╮
         //                     │
         //                     ▢ target
```

These docs were very helpful to understand which part of a connection the code relates to. Can we bring them back, please?

Reply:
Added docs with diagrams to the `layout` module, explaining the key concepts of the design.